### PR TITLE
feat: consolidate application state fetching

### LINKED ISSUE #18986: Refactor initial app state to parallel API requests
The  UI is architected in a pattern that does not issue API calls until previous API calls have been resolved. 
```
<Component#1>
    <Component#2 />
</Component#1>
```
and component#1 does not render component#2 until after all the API calls it makes are resolved. 
The biggest culprit here is the use of SpinnerContainer. which blocks the render of its children while it is loading. What happens is that we show one container's Spinner and then its child's spinner, and then its child's spinner.

### PR DESCRIPTION
**Pull request by drdelambre**
Closes #18986

instead of cascading blocking operations, lets grab them all at once!

### CODE REVIEW

File: ui/src/me/components/account/Settings.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ export class Settings extends PureComponent<StateProps, State> {
 }
 
 const mstp = ({me}: AppState) => ({
-  me,
+  me: me.resource.name,
```
Comment (ebb-tide) [inline ui/src/me/components/account/Settings.tsx:67]: I think this is supposed to be returning username?

File: ui/src/me/components/account/Settings.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ export class Settings extends PureComponent<StateProps, State> {
 }
 
 const mstp = ({me}: AppState) => ({
```
Comment (ebb-tide) [inline ui/src/me/components/account/Settings.tsx:66]:
```suggestion
const mstp = ({me}: AppState): StateProps => ({
```

File: ui/src/shared/containers/GetAppState.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      return {
+        flags: activeFlags(state),
+        flagStatus: state.flags.status || RemoteDataState.NotStarted,
+        orgStatus: state.resources.orgs.status || RemoteDataState.NotStarted,
+        meStatus: ignoreMe ? RemoteDataState.Done : state.me.status,
+      }
+    }
+  )
+
+  const loading = [orgStatus, meStatus, flagStatus].filter(
```
Comment (ebb-tide) [inline ui/src/shared/containers/GetAppState.tsx:50]: I wonder if `some` is more appropriate here?

File: ui/src/shared/containers/GetAppState.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        dispatch(notify(sessionTimedOut()))
+      }
+
+      history.replace(`/signin${returnTo}`)
+    }
+  }, [dispatch, meStatus])
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      // NOTE: after the initial load, this dispatch will update the loading state, which will cause a rerender
```
Comment (ebb-tide) [inline ui/src/shared/containers/GetAppState.tsx:105]: I didn't quite get this Will it rerender at each interval?

Comment (drdelambre) [reply ui/src/shared/containers/GetAppState.tsx:105]: it will without this variable

Comment (ebb-tide) [reply ui/src/shared/containers/GetAppState.tsx:105]: got it

**Comment by drdelambre** [comment]
i'll look into that

**Pull request merged**